File: market-domain/src/main/java/com/example/market/domain/trading/BidOwnershipViolation.java
package com.example.market.domain.trading;

import com.example.market.domain.shared.UserId;

/**
 * 다른 사용자의 Bid 를 조작하려고 할 때.
 */
public class BidOwnershipViolation extends RuntimeException {
    private static final long serialVersionUID = 1L;

    public BidOwnershipViolation(BidId bidId, UserId owner, UserId requestor) {
        super("bid " + bidId + " owned by " + owner + " — requestor " + requestor + " forbidden");
    }
}

File: market-domain/src/main/kotlin/com/example/market/domain/trading/BidOwnershipViolation.kt
package com.example.market.domain.trading

import com.example.market.domain.shared.UserId

/**
 * 다른 사용자의 Bid 를 조작하려고 할 때.
 */
class BidOwnershipViolation(
    bidId: BidId,
    owner: UserId,
    requestor: UserId,
) : RuntimeException(
    "bid $bidId owned by $owner — requestor $requestor forbidden",
) {
    companion object {
        private const val serialVersionUID: Long = 1L
    }
}
